### PR TITLE
Fix/1467 auth no prompt

### LINKED ISSUE #1467: lnurl-auth no longer logs you in automatically
### Describe the bug

lnurl-auth no longer logs you in automatically but prompts you every time.
if you're logged out on http://getalby.com or stacker.news or etc. and log back in Alby prompts you.

this is because it does not get updated here:  
https://github.com/getAlby/lightning-browser-extension/blob/master/src/extension/background-script/actions/allowances/update.ts

### Expected behavior

Auto-login should work

### PR DESCRIPTION
**Pull request by escapedcat**
### Describe the changes you have made in this PR

- fix non-prompt login 
- minor refactor

### Link this PR to an issue

Fixes #1467 

### Type of change (Remove other not matching type)

- `fix`: Bug fix (non-breaking change which fixes an issue)

### How has this been tested?

Manually login to SN with and without allowance

### CODE REVIEW

**Review by im-adithya — APPROVED** [review]
utACK

File: src/extension/background-script/actions/allowances/update.ts [hunk 1]
```diff
@@ -5,20 +5,25 @@ type OptionalPick<T, K extends keyof T> = { [P in K]?: T[P] };
...
   > = {};
-  if (Object.prototype.hasOwnProperty.call(message.args, "totalBudget")) {
+
+  if (message.args.totalBudget) {
     update.totalBudget = message.args.totalBudget;
     update.remainingBudget = message.args.totalBudget;
   }
-  if (Object.prototype.hasOwnProperty.call(message.args, "enabled")) {
+
+  if (message.args.enabled) {
```
Comment (bumi) [inline src/extension/background-script/actions/allowances/update.ts:20]: this breaks things. there is a reason why `Object.prototype.hasOwnProperty`
what values can `enabled` or `lnurlAuth` have?
try setting it to false.

**Review by bumi — CHANGES_REQUESTED** [review]

File: src/extension/background-script/actions/allowances/update.ts (hunk 1 above)
Comment (escapedcat) [reply src/extension/background-script/actions/allowances/update.ts]: Yes, thanks! Corrected and added a test

**Review by bumi — COMMENTED** [review]
this could have been DRYed up.

**Comment by escapedcat** [comment]
> this could have been DRYed up.

Happy for a hint

**Comment by bumi** [comment]
```
[a,b,c].forEach((attr) => { do things })
```

**Pull request merged**